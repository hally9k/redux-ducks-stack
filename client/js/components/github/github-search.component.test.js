import GithubSearch from 'components/github/github-search.component';
import React from 'react';
import { mount, shallow } from 'enzyme';

describe('components', () => {
    describe('GithubSearch', () => {
        it('should render the error message when given an error', () => {
            const props = {
                getGithubUser: jest.fn(),
                user: null,
                error: {
                    message: 'ajax error 404'
                }
            };
            const enzymeWrapper = mount(<GithubSearch {...props} />);

            expect(enzymeWrapper.find('label').text()).toBe('Github User: ');
            expect(enzymeWrapper.find('button').text()).toBe('Get User');
            expect(enzymeWrapper.find('div.error-block').exists()).toBe(true);
            enzymeWrapper.find('button').simulate('click');
            expect(props.getGithubUser).toHaveBeenCalled();
        });

        it('mounts the component', () => {
            const props = {
                getGithubUser: jest.fn(),
                user: null,
                error: {
                    message: 'ajax error 404'
                }
            };
            const enzymeWrapper = shallow(<GithubSearch {...props} />);
            expect(enzymeWrapper).toMatchSnapshot();
        });
    });
});
