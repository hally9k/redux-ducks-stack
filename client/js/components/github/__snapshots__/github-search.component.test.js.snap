// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components GithubSearch mounts the component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <label>
        Github User: 
        <input />
    </label>
    <button
        onClick={[Function]}
    >
        Get User
    </button>
    <ErrorBlock
        message="ajax error 404"
    />
</div>,
  "nodes": Array [
    <div>
      <label>
            Github User: 
            <input />
      </label>
      <button
            onClick={[Function]}
      >
            Get User
      </button>
      <ErrorBlock
            message="ajax error 404"
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <GithubSearch
        error={
                Object {
                        "message": "ajax error 404",
                      }
        }
        getGithubUser={[Function]}
        user={null}
/>,
      "_debugID": 12,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": GithubSearch {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "error": Object {
            "message": "ajax error 404",
          },
          "getGithubUser": [Function],
          "user": null,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 5,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <label>
                    Github User: 
                    <input />
          </label>
          <button
                    onClick={[Function]}
          >
                    Get User
          </button>
          <ErrorBlock
                    message="ajax error 404"
          />
</div>,
        "_debugID": 13,
        "_renderedOutput": <div>
          <label>
                    Github User: 
                    <input />
          </label>
          <button
                    onClick={[Function]}
          >
                    Get User
          </button>
          <ErrorBlock
                    message="ajax error 404"
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <GithubSearch
    error={
        Object {
            "message": "ajax error 404",
          }
    }
    getGithubUser={[Function]}
    user={null}
/>,
}
`;
